### PR TITLE
Migrate from Travis to Github Actions

### PR DESCRIPTION
**Pull request by jennydaman**
Both `nosetests` and CUBE integration tests are run

Regression: no more slack notifications (I don't think they were working in the first place, plus there was an exposed API key). It would be easy to put slack notifications back in if desired, see https://github.com/marketplace/actions/slack-notify

# Examples

- https://github.com/FNNDSC/cube-integration-action#example
- https://github.com/jennydaman/pfcon/actions

# Changes

- add `.github/workflows/ci.yml`
- delete `cube_before_install.sh`, Github's runners come preinstalled with `docker-compose`
- delete `cube_before_script.sh`, moved to [https://github.com/FNNDSC/cube-integration-action](cube-integration-action)
- delete `.travis.yml`

In summary, code can be reused more consistently thanks to [https://github.com/FNNDSC/cube-integration-action](cube-integration-action)

pfioh and pman can also benefit from [https://github.com/FNNDSC/cube-integration-action](cube-integration-action) using a similar `.github/workflows/ci.yml`

### CODE REVIEW

**Comment by junaruga** [comment]
@jennydaman Thank you for this work! It looks great!

> In summary, code can be reused more consistently thanks to https://github.com/FNNDSC/cube-integration-action

I can understand the commonly used code `uses: FNNDSC/cube-integration-action@v1` is beneficial for the maintainability of other repository's CIs. I am curious about any other benefits of Github Actions. Because Travis CI also has this kind of feature.
Did we have the discussion about migrating Travis to GitHub Actions in the past on Slack or somewhere?

I just find a typo in your comment. The used URL `https://github.com/FNNDSC/pfcon/pull/cube-integration-action` as a link is wrong.

CC: @BillRainford working on https://github.com/FNNDSC/ChRIS_contributor-support/issues/5 , and @mamurak working on https://github.com/FNNDSC/pfcon/pull/76 modifying `.travis.yml` there.

**Comment by mamurak** [comment]
I think it's worthwile to look into this. The current integration testing contains quite a bit of code to set up the environment properly, so it would be good to simplify the CI.
Two comments:
* The current service-internal `nosetests` are not needed in the CI once they're automatically executed during the container build step.
* I'm wondering whether it makes sense to use the [service containers feature](https://docs.github.com/en/free-pro-team@latest/actions/guides/about-service-containers) of Github Actions here.

**Comment by junaruga** [comment]
> Did we have the discussion about migrating Travis to GitHub Actions in the past on Slack or somewhere?

I found it on the `#devops` channel.
https://fnndsc.slack.com/archives/C019ST4CLF9/p1606872062001000

**Comment by jennydaman** [comment]
@mamurak I took a look at the services container feature, I really wanted to use it but decided against it.

- cannot maintain common config between `services` and `docker-compose.yml`
- I did not find documentation on being able to bootstrap a job's services within an `action.yml` (so no `uses: cube-integration-action@v1`
- CUBE provisioning actually involves complicated steps (join swarm, export STOREBASE), using `services` wouldn't be of much help 

There are a few benefits to Github Actions:

- single platform is convenient
- migration to travis-ci.com averted
- new pricing model for travis-ci.com is affecting other open source teams, it could also affect us

There's no urgent need to migrate _everything_ to Github Actions but I think porting our new efforts on Github Actions will be more maintainable.

**Comment by junaruga** [comment]
> migration to travis-ci.com averted

Does someone (Travis?) automatically migrate our current travis-ci.org to travis-ci.com? Could you explain more about the context?

> new pricing model for travis-ci.com is affecting other open source teams, it could also affect us

Why did the new pricing model for travis-ci.com affect us using travis-ci.org?

**Comment by jennydaman** [comment]
@junaruga

- travis-ci.org is shutting down in 29 days.
- https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
- migration to travis-ci.com from travis-ci.org should not actually be rough though the disruption is a good opportunity to review our pipelines anyways, which are old. Currently there is no single source of truth about how to start CUBE for CI. Our `.travis.yml` for pfcon, pman, and pfioh inappropriately use build matrix environment variables to do what jobs are for. We also have a private API key for slack exposed in all of them.

**Comment by junaruga** [comment]
OK. Thanks for the explanation. I agree now it's good time to review our pipelines.
I also found this article.
https://www.theregister.com/2020/11/02/travis_ci_pricng/

**Comment by junaruga** [comment]
After reading the following page in Travis Community, in my impression, we have to migrate Travis to alternative such as GitHub Actions for all the repositories of FNNDSC.
https://travis-ci.community/t/org-com-migration-unexpectedly-comes-with-a-plan-change-for-oss-what-exactly-is-the-new-deal/10567/14

**Comment by jennydaman** [comment]
Yikes, I simply assumed that "1,000 minutes" refreshed monthly. So it is really just a trial and not a free tier?

Travis-CI is asking OSS to email support to request more build limit credit allotment, for free, However many OSS projects are abandoning ship altogether with Travis.

**Comment by junaruga** [comment]
> Yikes, I simply assumed that "1,000 minutes" refreshed monthly. So it is really just a trial and not a free tier?

(Maybe) yes. I think it's a trial not a free tier, and as far as I read the thread in the Travis Community and seeing the following part in their announce, the 10K credit is one time deal. But we will see it next month..

> The upcoming pricing change will affect those of you who are:
> ...
> Building on a public repositories only
> ...
> For those of you who have been building on public repositories (on travis-ci.com, with no paid subscription), we will upgrade you to our trial (free) plan with a 10K credit allotment (which allows around 1000 minutes in a Linux environment).

[1] https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

**Comment by junaruga** [comment]
> Travis-CI is asking OSS to email support to request more build limit credit allotment, for free, However many OSS projects are abandoning ship altogether with Travis.

I am not familiar with many OSS projects situation, but when I just saw the Ruby project today, they mainly moved their CI cases to GitHub Actions keeping only non-x86_64 cases on Travis last month.
